tools/ci_build/github/apple/test_ios_packages.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python3
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import argparse
+import os
+import pathlib
+import shutil
+import subprocess
+
+SCRIPT_DIR = os.path.dirname(os.path.realpath(__file__))
+REPO_DIR = os.path.normpath(os.path.join(SCRIPT_DIR, "..", "..", "..", ".."))
+
+
+def _test_ios_packages(args):
+    # check if CocoaPods is installed
+    if shutil.which('pod') is None:
+        if args.fail_if_cocoapods_missing:
+            raise ValueError('CocoaPods is required for this test')
+        else:
+            print('CocoaPods is not installed, ignore this test')
+
+    # Now we need to create a zip file contains the framework and the podspec file, both of these 2 files
+    # should be under the c_framework_dir
+    c_framework_dir = args.c_framework_dir.resolve()
+    if not c_framework_dir.is_dir():
+        raise FileNotFoundError('c_framework_dir {} is not a folder.'.format(c_framework_dir))
+
+    framework_path = os.path.join(c_framework_dir, 'onnxruntime.framework')
+    if not pathlib.Path(framework_path).exists():
+        raise FileNotFoundError('{} does not have onnxruntime.framework'.format(c_framework_dir))
+
+    # create a temp folder under repo dir
+    import tempfile
+    with tempfile.TemporaryDirectory() as temp_dir:
+        # temp_dir = '/Users/gwang/temp/eee'
+
+        # create a zip file contains the framework
+        local_pods_dir = os.path.join(temp_dir, 'local_pods')
+        os.makedirs(local_pods_dir, exist_ok=True)
+        # shutil.make_archive require target file as full path without extension
+        zip_base_filename = os.path.join(local_pods_dir, 'OnnxRuntimeBase')
+        zip_file_path = zip_base_filename + '.zip'
+        shutil.make_archive(zip_base_filename, 'zip', root_dir=c_framework_dir, base_dir='onnxruntime.framework')
+
+        # copy the test project to the temp_dir
+        test_proj_path = os.path.join(REPO_DIR, 'onnxruntime', 'test', 'platform', 'ios', 'ios_package_test')
+        target_proj_path = os.path.join(temp_dir, 'ios_package_test')
+        if pathlib.Path(target_proj_path).exists():
+            shutil.rmtree(target_proj_path)
+        shutil.copytree(test_proj_path, target_proj_path)
+
+        # update the podspec to point to the local framework zip file
+        local_podspec_path = os.path.join(target_proj_path, 'OnnxRuntimeBase.podspec')
```

Comment:
>OnnxRuntimeBase.podspec [](http://example.com/codeflow?start=61&length=23)

maybe call it OnnxRuntimeBase.podspec.template or something to indicate that it is not an actual podspec, but intended to have some placeholders replaced.